Scrapper-Service/plugins/guide2research.py

```diff
@@ -0,0 +1,354 @@
+import logging
+from datetime import datetime as dt
+from typing import Dict, List
+
+import requests
+from bs4 import BeautifulSoup as bs
+
+from commons import PageParsingError, Scrapper
+from datamodels import Conference, Metadata
+class Guide2ResearchScrapper(Scrapper):
+    def __init__(self, **config):
+        super().__init__(context_name=__name__, **config)
+        self.base_address = "http://www.guide2research.com/"
+        self.top_conf_address = self.base_address + "topconf/"
+        self.all_conf_address = self.base_address + "conferences/"
+        self.site_name = "guide2research"
+
+    def parse_action(self):
+        self.extract_and_push()
+
+    def extract_and_push(self):
+        """
+        Main calling method for g2r scraper
+        Collects the links of all top conferences available
+        and posts them to db
+        """
+        top_conference_links = self._get_conferences_list(which='top')
+        all_conference_links = self._get_conferences_list(which='all')
+        n_top_confs = len(top_conference_links)
+        n_all_confs = len(all_conference_links)
+        for i,link in enumerate(top_conference_links):
+            try:
+                conf_data = self._parse_top_conference(link=link)
+                self.logger.info(f"Conference {i}/{n_top_confs} Top Conferences added")
+                self.push_todb(conf_data)
+            except Exception as e:
+                self.logger.error(
+                    f"Error while parsing page {link}, find full trace {e}")
+        for i,link in enumerate(all_conference_links):
+            try:
+                conf_data = self._parse_all_conference(link=link)
+                self.logger.info(f"Conference {i}/{n_all_confs} All Conferences added")
+                self.push_todb(conf_data)
+            except Exception as e:
+                self.logger.error(
+                    f"Error while parsing {link}, full trace {e}"
+                )
+
+    def _get_page(self, base_address: str) -> bytes:
+        """
+        Takes in page address and returns html content
+
+        Args:
+        ---
+        base_address: string
+
+        Returns:
+        ---
+        page content: bytes
+
+        """
+        page = requests.get(base_address)
```

Comment:
* Since u will be using the self.get_page from the inherited class , only to add . always specify requests.get() wait timeout. Read about it in docs. Not specifying the timeout parameter can make it wait indefinitely by default. it was a critical bug that was addressed already. Hence advicing to use preimplemented methods.

Reply:
Done